profiles/killbill/src/main/java/org/killbill/billing/server/log/obfuscators/Obfuscator.java

```diff
@@ -17,90 +17,10 @@
-
-    protected static final char PAD_CHAR = '*';
-
-    public abstract String obfuscate(final String originalString, final ILoggingEvent event);
-
-    protected String obfuscate(final String originalString, final Iterable<Pattern> patterns, final ILoggingEvent event) {
-        final StringBuilder obfuscatedStringBuilder = new StringBuilder(originalString);
-
-        for (final Pattern pattern : patterns) {
-            int currentOffset = 0;
-            // Create a matcher with a copy of the current obfuscated String
-            final Matcher matcher = pattern.matcher(obfuscatedStringBuilder.toString());
-            while (matcher.find()) {
-                for (int groupNb = 1; groupNb <= matcher.groupCount(); groupNb++) {
-                    final String confidentialData = matcher.group(groupNb);
-
-                    if (shouldObfuscate(confidentialData, event)) {
-                        final String obfuscatedConfidentialData = obfuscateConfidentialData(confidentialData);
-
-                        obfuscatedStringBuilder.replace(currentOffset + matcher.start(groupNb), currentOffset + matcher.end(groupNb), obfuscatedConfidentialData);
-
-                        // The original String is modified in place, which will confuse the Matcher if it becomes bigger
-                        if (obfuscatedConfidentialData.length() > confidentialData.length()) {
-                            currentOffset += obfuscatedConfidentialData.length() - confidentialData.length();
-                        }
-                    }
-                }
-            }
-        }
-
-        return obfuscatedStringBuilder.toString();
-    }
-
-    private boolean shouldObfuscate(final String confidentialData, final ILoggingEvent event) {
-        return !isProfilingHeader(confidentialData, event);
-    }
-
-    // Huge hack to avoid obfuscating the "name" in the X-Killbill-Profiling-Resp json. Unfortunately, we can't simply
-    // filter-out c.s.j.a.c.filter.LoggingFilter because we do want to obfuscate requests (in case sensitive data is passed as
-    // query parameters, e.g. in plugin properties)
-    private boolean isProfilingHeader(final String confidentialData, final ILoggingEvent event) {
-        if (!LOGGING_FILTER_NAME.equals(event.getLoggerName())) {
-            return false;
-        }
 
-        for (final ProfilingFeatureType profileType : ProfilingFeatureType.values()) {
-            // See ProfilingDataItem#getKey
-            if (confidentialData.startsWith("\"" + profileType.name() + ":")) {
-                return true;
-            }
-        }
-        return false;
-    }
+public interface Obfuscator {
 
-    private String obfuscateConfidentialData(final CharSequence confidentialSequence) {
-        return obfuscateConfidentialData(confidentialSequence, null);
-    }
+    String obfuscate(final String originalString, final ILoggingEvent event);
```

Comment:
If the goal is to share this interface (e.g. with plugins), I think it would make sense to remove the dependency on `Logback`. We could have our own, lightweight, `LoggingEvent` class?